From 6a513a8febf9de2746e063cedab8651b8add1ef4 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Wed, 23 Feb 2022 17:37:47 -0500
Subject: [PATCH 01/19] Let system override ro.apex.updatable

APEX are broken because of a kernel bug in Android 10 devices
So we have system set ro.apex.updatable = false
However, vendor can override system, which prevents us from setting this
to false
So, ignore the override policy for this prop
---
 init/property_service.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index 6f8da33fc..be3841686 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -727,8 +727,12 @@ static void LoadProperties(char* data, const char* filter, const char* filename,
                 } else if (it->second != value) {
                     LOG(WARNING) << "Overriding previous property '" << key << "':'" << it->second
                                  << "' with new value '" << value << "'";
-                    it->second = value;
-                }
+                    if(strcmp("ro.apex.updatable", key) == 0) {
+                        LOG(WARNING) << "... Ignored";
+                    } else {
+                        it->second = value;
+                    }
+		}
             } else {
                 LOG(ERROR) << "Do not have permissions to set '" << key << "' to '" << value
                            << "' in property file '" << filename << "': " << error;
-- 
2.37.1 (Apple Git-137.1)

